cmd/dfdaemon/app/root.go

```diff
@@ -0,0 +1,107 @@
+package app
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/dragonflyoss/Dragonfly/cmd/dfdaemon/options"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/initializer"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	opt           = &options.Options{}
+	defaultDFRepo = filepath.Join(os.Getenv("HOME"), ".small-dragonfly/dfdaemon/data/")
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "dfdaemon",
+	Short: "The dfdaemon is a proxy that intercepts image download requests.",
+	Long:  "The dfdaemon is a proxy between container engine and registry used for pulling images.",
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Can we use `RunE` which can propagate errors to `Execute`?

Reply:
Yeah, got it.